ignite/services/plugin/plugin_test.go

```diff
@@ -353,6 +362,104 @@ func TestPluginLoad(t *testing.T) {
 	}
 }
 
+func TestPluginLoadSharedHost(t *testing.T) {
+	// wd, err := os.Getwd()
+	// require.NoError(t, err)
+
+	// scaffoldPlugin runs Scaffold and updates the go.mod so it uses the
+	// current ignite/cli sources.
+	scaffoldPlugin := func(t *testing.T, dir, name string) string {
+		require := require.New(t)
+		path, err := Scaffold(dir, name, true)
+		require.NoError(err)
+		// We want the scaffolded plugin to use the current version of ignite/cli,
+		// for that we need to update the plugin go.mod and add a replace to target
+		// current ignite/cli
+		gomod, err := gomodule.ParseAt(path)
+		require.NoError(err)
+		// use GOMOD env to get current directory module path
+		modpath, err := gocmd.Env(gocmd.EnvGOMOD)
+		require.NoError(err)
+		modpath = filepath.Dir(modpath)
+		gomod.AddReplace("github.com/ignite/cli", "", modpath, "")
+		// Save go.mod
+		data, err := gomod.Format()
+		require.NoError(err)
+		err = os.WriteFile(filepath.Join(path, "go.mod"), data, 0o644)
+		require.NoError(err)
+		return path
+	}
+
+	tests := []struct {
+		name          string
+		buildPlugin   func(t *testing.T) Plugin
+		expectedError string
+		instances     int
+	}{
+		{
+			name: "ok: from local sharedhost is on 1",
+			buildPlugin: func(t *testing.T) Plugin {
+				path := scaffoldPlugin(t, t.TempDir(), "github.com/foo/bar-1")
+				return Plugin{
+					Plugin:     pluginsconfig.Plugin{Path: path},
+					srcPath:    path,
+					binaryName: "bar-1",
+				}
+			},
+			instances: 1,
+		},
+		{
+			name: "ok: from local sharedhost is on 2",
+			buildPlugin: func(t *testing.T) Plugin {
+				path := scaffoldPlugin(t, t.TempDir(), "github.com/foo/bar-2")
+				return Plugin{
+					Plugin:     pluginsconfig.Plugin{Path: path},
+					srcPath:    path,
+					binaryName: "bar-2",
+				}
+			},
+			instances: 2,
+		},
+		{
+			name: "ok: from local sharedhost is on 4",
+			buildPlugin: func(t *testing.T) Plugin {
+				path := scaffoldPlugin(t, t.TempDir(), "github.com/foo/bar-4")
+				return Plugin{
+					Plugin:     pluginsconfig.Plugin{Path: path},
+					srcPath:    path,
+					binaryName: "bar-4",
+				}
+			},
+			instances: 4,
+		},
```

Comment:
Only the number of instances is changing between those cases right?

Reply:
yes, all other properties are the same.

Reply:
modified the `TestPluginLoadSharedHost` tests a bit to take `sharesHost` as a property to the struct so scaffolding the plugin options can be passed in. added some checks to exercise behaviors when the `SharedHost` flag is false in the plugin manifest to assure correct behaviors for both scenarios of plugin instantiation.

Reply:
While refactoring that test, I found an issue: the `scaffoldPlugin` was invoked for each instance, which affect the plugin path (different temp dir), as a result the `CheckPluginConfCache` was always returning false for all instances.

This divides the test duration by the number of instances, because `scaffoldPlugin` is a costly operation.

In your version, the test was OK because you were always asserting `plugins[0].isHost` instead of `plugins[i].isHost`.

In addition, when `sharedHost` was disabled, the plugins were not killed. I fixed that also by using a `defer`.

See c1e9351c